documentation/src/main/asciidoc/topics/protostream.adoc

```diff
@@ -0,0 +1,163 @@
+[[protostream]]
+= ProtoStream
+
+link:https://github.com/infinispan/protostream[ProtoStream] is a serialization library based on link:https://developers.google.com/protocol-buffers[Protocol Buffers].
+The link:https://github.com/infinispan/protostream[ProtoStream] library, is responsible for encoding/decoding Java objects
+to/from the Protobuf format and is used extensively throughout {brandname}.
+
+This chapter provides a brief overview of the Protobuf and ProtoStream concepts, before detailing how ProtoStream can be
+used to serialize your Java objects.
+
+== Concepts
+=== .proto Files
+NOTE: Protocol Buffers is a broad subject, we will not detail it here in great detail, so please consult the
+link:https://developers.google.com/protocol-buffers/docs/overview[Developer Guide] for an in-depth
+explanation of the encoding format and best practices.
+
+Protocol Buffers, Protobuf for short, provide a platform independent encoding format that utilises .proto schema files to
+provide a structured representation of entities that can be easily evolved over time whilst maintaining backwards compatibility.
+
+Protobuf is all about structured data, so the first thing todo is to define the structure of your data. This is
+accomplished by declaring Protobug message types in .proto files, as shown in the example below.
+
+[[book_sample_proto]]
+[source,proto]
+.library.proto
+----
+include::config_examples/library.proto[]
+----
+
+In this example .proto file, we define an entity (message type in Protobuf speak) named _Book_, which is contained in the
```

Comment:
Not sure about the style guidelines, but I feel the .proto extension should be surrounded by _

Reply:
asciidoc markup conventions use ` to surround filenames 

https://redhat-documentation.github.io/asciidoc-markup-conventions/

to the letter [filename]`.proto` would be correct but imo just go for `.proto`